### PR TITLE
fix race condition in hybrid overlay DRIP alloc

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
we need to ensure that we only call allocateHybridOverlayDRIP() in syncPods() when there is an existing hybrid overlay distributed router ip address on the node.

the initial allocation of the hybrid overlay distributed router ip should be made during the call to allocateHybridOverlayDRIP() during the addUpdateNodeEvent(). The call to allocateHybridOverlayDRIP() is in syncPods() to ensure that on restart the annotated hybrid overlay distributed router ip is allocated before processing pod events so it is reserved. without the check for the presence of the annotation there is a good chance that the syncPods() will call allocateHybridOverlayDRIP() with a stale node object and cause a reallocation of the hybrid overlay distributed router ip without changing associated the MAC address or freeing the previous.

Signed-off-by: Jacob Tanenbaum <jtanenba@redhat.com>

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/ovn/pods.go [hunk 1]
```diff
@@ -74,8 +75,10 @@ func (oc *DefaultNetworkController) syncPods(pods []interface{}) error {
 			return fmt.Errorf("failed to get nodes: %v", err)
 		}
 		for _, node := range nodes {
-			if err := oc.allocateHybridOverlayDRIP(node); err != nil {
-				return fmt.Errorf("cannot allocate hybridOverlay DRIP on node %s (%v)", node.Name, err)
+			if _, ok := node.Annotations[hotypes.HybridOverlayDRIP]; ok {
+				if err := oc.allocateHybridOverlayDRIP(node); err != nil {
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/pods.go:82]: done

**Comment by dcbw** [comment]
DUal-conversion jobs seem to have port conflict issues that are unrelated to this PR:

```
ovn-control-plane/containers/kube-scheduler-ovn-control-plane_kube-system_kube-scheduler-4eda0aeef2fd6a5f24e121ae4a6568c04f4e492c2134a5f78ef2fbe15157bcc3.log:2023-01-17T21:24:58.677154881Z stderr F I0117 21:24:58.677114       1 scheduler.go:351] "Unable to schedule pod; no fit; waiting" pod="dualstack-3798/netserver-2" err="0/3 nodes are available: 2 node(s) didn't match Pod's node affinity/selector, 3 node(s) didn't have free ports for the requested pod ports. preemption: 0/3 nodes are available: 1 No preemption victims found for incoming pod, 2 Preemption is not helpful for scheduling."
ovn-control-plane/containers/kube-scheduler-ovn-control-plane_kube-system_kube-scheduler-4eda0aeef2fd6a5f24e121ae4a6568c04f4e492c2134a5f78ef2fbe15157bcc3.log:2023-01-17T21:24:58.677983991Z stderr F I0117 21:24:58.677953       1 schedule_one.go:847] "Updating pod condition" pod="dualstack-3798/netserver-2" conditionType=PodScheduled conditionStatus=False conditionReason="Unschedulable"
```

For example, netserver pods fail to start due to port conflicts which causes some tests to fail.

**Pull request merged**